### DISCUSSION TITLE
Is there any way to call winget programatically from C#?

### DISCUSSION THREAD

**Discussion by fgimian** [microsoft/winget-cli#4469] [General] [answered]
Hey there folks, love your work on winget.

I know that there is the PowerShell module available which I have played with, but I was wondering if there was any way to call into winget via C# directly programatically (without going via a PowerShell interpreter)?

Looking at the PowerShell module, it seems that maybe the `PackageManagerWrapper` class could do everything one would need via C# directly, but I don't see a NuGet package published that I can install to get access to this class.

Is there any way to do this presently or am I missing anything?

Cheers
Fotis

**Comment by fgimian** [comment]
I've been doing a lot more work on this and have it all working. The best reference is the implementation [here](https://github.com/microsoft/devhome/tree/main/tools/SetupFlow/DevHome.SetupFlow.Common/WindowsPackageManager).

The DevHome codebase has examples of using the factory to create various objects and work with the WinGet package manager via C#.

Cheers
Fotis

**Comment by marticliment** [comment, marked as answer]
I managed to isolate a minimal working example here: https://github.com/marticliment/WinGet-API-from-CSharp

Perhaps this may be useful to you

**Reply by fgimian** [reply]
Thank you so much!
